### PR TITLE
Disable OPENTHREAD_CONFIG_NCP_HDLC_ENABLE config when OT_NCP_SPI is set.

### PR DESCRIPTION
**Pull request by parag-silabs**

### CODE REVIEW

**Review by lmnotran — APPROVED** [review]

**Review by romacdon — APPROVED** [review]
LGTM

File: src/lib/spinel/CMakeLists.txt [hunk 1]
```diff
@@ -31,13 +31,19 @@ add_library(openthread-spinel-rcp)
 
 target_compile_definitions(openthread-spinel-ncp PRIVATE
     OPENTHREAD_FTD=1
-    OPENTHREAD_CONFIG_NCP_HDLC_ENABLE=1
     PUBLIC OPENTHREAD_SPINEL_CONFIG_OPENTHREAD_MESSAGE_ENABLE=1
 )
 
+if (OT_NCP_SPI)
+    target_compile_definitions(openthread-spinel-ncp PRIVATE OPENTHREAD_CONFIG_NCP_HDLC_ENABLE=0)
```
Comment (suveshpratapa) [inline src/lib/spinel/CMakeLists.txt:38]: Any reason these lines are duplicated twice here?

**Review by suveshpratapa — CHANGES_REQUESTED** [review]

File: src/lib/spinel/CMakeLists.txt (hunk 1 above)
Comment (parag-silabs) [reply src/lib/spinel/CMakeLists.txt:38]: One is `**ncp**` and second is `**rcp**`.

Comment (romacdon) [reply src/lib/spinel/CMakeLists.txt:38]: They're not notice ncp vs rcp

Comment (suveshpratapa) [reply src/lib/spinel/CMakeLists.txt:38]: @parag-silabs Yes, seems to be a typo then.  Needs change for one of the lines in both cases to say `rcp`

Comment (suveshpratapa) [reply src/lib/spinel/CMakeLists.txt:38]: My bad.

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by bertoldi-silabs — APPROVED** [review]

**Pull request closed without merging**